File: Tercera Entrega/sdi3-37.Cli-SOAP/src/com/sdi/presentation/admin/action/ListarUsuariosAction.java
package com.sdi.presentation.admin.action;

import java.util.List;

import com.sdi.ws.AdminService;
import com.sdi.ws.EjbAdminServiceService;
import com.sdi.ws.UserInfo;
import alb.util.console.Console;
import alb.util.menu.Action;

public class ListarUsuariosAction implements Action {	
	
	public ListarUsuariosAction() {}

	@Override
	public void execute() throws Exception {
		try {
			AdminService aService = new EjbAdminServiceService().getAdminServicePort();
			
			List<UserInfo> usuarios = aService.listAllUsers();
			
			print(usuarios);
			
		}
		catch(Exception e) {
			e.printStackTrace();
		}
	}
	
	private void print(List<UserInfo> userInfo) {		
		Console.printf("%-20s%-20s%-20s%-20s%-27s%-20s%-20s\n","LOGIN","EMAIL","ESTADO","COMPLETADAS", "COMPLETADAS_RETRASADAS", "PLANEADAS", "NO_PLANEADAS");
		for(UserInfo u: userInfo) {
			if(!u.getUser().isIsAdmin()) {				
				Console.printf("%-20s", u.getUser().getLogin());
				Console.printf("%-20s", u.getUser().getEmail());
				Console.printf("%-25s", String.valueOf(u.getUser().getStatus()));
				Console.printf("%-25s",u.getTareasCompletadas());
				Console.printf("%-21s",u.getTareasCompletadasRetrasadas());
				Console.printf("%-22s",u.getTareasPlanificadas());
				Console.printf("%-20s\n", u.getTareasSinPlanificar());				
			}			
		}
	}
}

File: Tercera Entrega/sdi3-37.Cli-SOAP-C_Sharp/sdi3-37.Cli-SOAP-C_Sharp/action/ListarUsuariosAction.cs
using System;
using sdi3_37.Cli_SOAP_C_Sharp.SOAP;

namespace sdi3_37.Cli_C_Sharp
{
    class ListarUsuariosAction : Action
    {      
        public ListarUsuariosAction()
        {

        }

        public void execute()
        {
            Console.WriteLine("---------------Listar usuarios---------------");

            try
            {
                AdminService aService = new AdminServiceClient();

                userInfo[] usuarios = aService.listAllUsers(new listAllUsers()).@return;

                print(usuarios);

            } catch(Exception e)
            {
                Console.WriteLine("\tError grave");
            }
            
            Console.WriteLine("---------------------------------------------");
        }

        private void print (userInfo[] usuarios)
        {
            Console.WriteLine(String.Format("{0,-20}{1,-20}{2,-20}{3,-20}{4,-27}{5,-20}{6,-20}", "LOGIN", "EMAIL", "ESTADO", "COMPLETADAS", "COMPLETADAS_RETRASADAS", "PLANIFICADAS", "NO_PLANIFICADAS"));

            foreach (var u in usuarios)
            {
                if (!u.user.isAdmin)
                {
                    Console.Write(String.Format("{0,-20}", u.user.login));
                    Console.Write(String.Format("{0,-20}", u.user.email));
                    Console.Write(String.Format("{0,-25}", u.user.status));
                    Console.Write(String.Format("{0,-25}", u.tareasCompletadas));
                    Console.Write(String.Format("{0,-21}", u.tareasCompletadasRetrasadas));
                    Console.Write(String.Format("{0,-22}", u.tareasPlanificadas));
                    Console.WriteLine(String.Format("{0,-20}", u.tareasSinPlanificar));
                }
            }
        }
    }
}
